Fill.py

```diff
@@ -232,13 +232,14 @@ def accessibility_corrections(world: MultiWorld, state: CollectionState, locatio
             if location in state.events:
                 state.events.remove(location)
             locations.append(location)
-
-    if pool:
+    if pool and locations:
+        locations.sort(key=lambda loc: loc.progress_type != LocationProgressType.PRIORITY)
         fill_restrictive(world, state, locations, pool)
 
 
 def inaccessible_location_rules(world: MultiWorld, state: CollectionState, locations):
-    maximum_exploration_state = sweep_from_pool(state, [])
+    maximum_exploration_state = state.copy()
+    maximum_exploration_state.sweep_for_events()
```

Comment:
While I agree that the for loop is useless, sweep_from_pool does have an empty default for itempool. I think leaving it as single line is cleaner and more consistent than doing an explicit copy and sweep.
```suggestion
    maximum_exploration_state = sweep_from_pool(state)
```